src/plugins/workspace/server/plugin.ts

```diff
@@ -64,12 +70,80 @@ export class WorkspacePlugin implements Plugin<WorkspacePluginSetup, WorkspacePl
     });
   }
 
+  private async setupPermission(
+    core: CoreSetup,
+    { applicationConfig }: AppPluginSetupDependencies
+  ) {
+    this.permissionControl = new SavedObjectsPermissionControl(this.logger);
+
+    core.http.registerOnPostAuth(async (request, response, toolkit) => {
+      let groups: string[];
+      let users: string[];
+
+      // There may be calls to saved objects client before user get authenticated, need to add a try catch here as `getPrincipalsFromRequest` will throw error when user is not authenticated.
+      try {
+        ({ groups = [], users = [] } = this.permissionControl!.getPrincipalsFromRequest(request));
+      } catch (e) {
+        return toolkit.next();
+      }
+      if (groups.length === 0 && users.length === 0) {
+        updateWorkspaceState(request, {
+          isDashboardAdmin: false,
+        });
```

Comment:
Why don't we use `updateDashboardAdminStateForRequest` method here?

Reply:
If users/groups is null, we can prevent the program from running down in advance. Meanwhile,  configGroups/configUsers has not been obtained.